src/Main.hh

```diff
@@ -256,6 +263,12 @@ private:
         std::string command = get_command();
         if (this->wrapper.length())
             command = this->wrapper+" \""+command+"\"";
+
+		// --run-command switch
+		if (this->run_command.length()){
+			command = replace(this->run_command, this->run_command_token, command);
+		}
```

Comment:
I think command should be shell-quoted here, so that the command always gets the command line as a single argument? Not sure, really.

Reply:
I'd say that totally depends on the end user. They can perfectly choose to shell quote the `{}`

Reply:
One the other hand, if  we do shell-quote by default and the user then - by standard practice - quotes again, it would be something like `"{}"` -> `""cmd""` which would completely defy the quotes. I guess the current approach should suffice. Let me know what you think.

Reply:
I'm only familiar with the {} syntax from `find(1)` - there are likely others, which probably behave differently - which escapes the file name (e.g .`find -exec strace -e execve echo {} \;` with some files with spaces in their names will look like this: `execve("/usr/bin/echo", ["echo", "./foo bar baz"], ...`, so the {} becomes a single argument).

Reply:
Added a new commit to resolve this issue. This is as far I could think which would maintain $SHELL usage and won't break with weird filenames (ones with quotes in them)